test/unit3/test_magic_mock.py

```diff
@@ -0,0 +1,352 @@
+__author__ = 'Duncan Richards'
+import unittest, math
+from unittest.mock import MagicMock, call, _magics
+
+
+
+class TestMockingMagicMethods(unittest.TestCase):
```

Comment:
Thanks for adding these extra tests

This seems to be a mixture of cpython tests and custom tests?

For tests here could we do the following:
- Take the `testsmagicmethods.py` file from cpython (which is the test file that this is based on)
- copy the file exactly (same name too)
- comment out tests that are not passing/not appropriate and make any other necessary tweaks

Create a separate file for custom tests, and name it appropriately e.g. `test_skulpt_magic_mock.py` (or can keep it in the same file but create a separate unittest.TestCase class)

This makes it easier to distinguish what is custom and what is from cpython for future developers.